python/version_details/version_details.py

```diff
@@ -370,6 +371,23 @@ def _set_notes_are_selectable(self, state):
         _set_notes_are_selectable,
     )
 
+    def _get_pre_submit_callback(self):
+        """
+        The pre-submit callback function, if one is registered. If so, this
+        Python callable will be run prior to Note or Reply submission, and
```

Comment:
Am I right thinking that if we changed the special in-line reply implementation that we have over to use a noteinput widget instead like suggested earlier, this callback could run for updates as well as other operations? Seems like another reason why that approach is beneficial.